benchmarks/segmentation_layers.py

```diff
@@ -1,175 +1,319 @@
-    return _cross_entropy_nt
-
-#
-# dropout
-#
-def dropout_tensor_tensors_same_size():
-    inputs = [torch.randn(C, H, W) for _ in range(N)]
-    a = torch.stack(inputs)
-
-    def _dropout_tensor():
-        torch.nn.functional.dropout(a)
-    return _dropout_tensor
-
-def dropout_nt_tensors_same_size():
-    inputs = [torch.randn(C, H, W) for _ in range(N)]
-    nt = nestedtensor.nested_tensor(inputs)
-    def _dropout_nt():
-        torch.nn.functional.dropout(nt)
-    
-    return _dropout_nt
-
-#
-# interpolate
-#
-def interpolate_tensor_tensors_same_size():
-    inputs = [torch.randn(C, H, W) for _ in range(N)]
-    a = torch.stack(inputs)
-
-    def _interpolate_tensor():
-        torch.nn.functional.interpolate(a, inputs[0].unsqueeze(0).shape[-2])
-    return _interpolate_tensor
-
-def interpolate_nt_tensors_same_size():
-    inputs = [torch.randn(C, H, W) for _ in range(N)]
-    nt = nestedtensor.nested_tensor(inputs)
-    def _interpolate_nt():
-        torch.nn.functional.interpolate(nt)
-    
-    return _interpolate_nt
+class SegLayersBenchMark(object):
+    def __init__(self, args):
+        assert len(args) == 8
+        print("Called with args: ")
+        
+        self.N = int(args[0])
+        print("N = ", self.N)
 
+        self.C = int(args[1])
+        print("C = ", self.C)
 
-if __name__ == "__main__":
-    print(utils.benchmark_fn(relu_tensor_tensors_same_size(), warmup=2.0))
-    print(utils.benchmark_fn(relu_nt_tensors_same_size(), warmup=2.0))
-    print(utils.benchmark_fn(conv2d_tensor_tensors_same_size(), warmup=2.0))
-    print(utils.benchmark_fn(conv2d_nt_tensors_same_size(), warmup=2.0))
-    print(utils.benchmark_fn(batch_norm_tensor_tensors_same_size(), warmup=2.0))
-    print(utils.benchmark_fn(batch_norm_nt_tensors_same_size(), warmup=2.0))
-    print(utils.benchmark_fn(max_pool2d_tensor_tensors_same_size(), warmup=2.0))
-    print(utils.benchmark_fn(max_pool2d_nt_tensors_same_size(), warmup=2.0))
-    print(utils.benchmark_fn(cross_entropy_tensor_tensors_same_size(), warmup=2.0))
-    print(utils.benchmark_fn(cross_entropy_nt_tensors_same_size(), warmup=2.0))
-    print(utils.benchmark_fn(dropout_tensor_tensors_same_size(), warmup=2.0))
-    print(utils.benchmark_fn(dropout_nt_tensors_same_size(), warmup=2.0))
-    print(utils.benchmark_fn(interpolate_tensor_tensors_same_size(), warmup=2.0))
-    print(utils.benchmark_fn(interpolate_nt_tensors_same_size(), warmup=2.0))
+        self.H = int(args[2])
+        print("H = ", self.H)
+
+        self.W = int(args[3])
+        print("W = ", self.W)
+        
+        self.var_h = int(args[4])
+        print("var_h = ", self.var_h)
+
+        self.var_w = int(args[5])
+        print("var_w = ", self.var_w)
+        
+        self.seed = int(args[6])
+        print("seed = ", self.seed)
+
+        self.warmup = float(args[7])
+        print("warmup = ", self.warmup)
+        
+        self.conv2d = torch.nn.Conv2d(self.C, 3, kernel_size= (1, 1), bias=False)
```

Comment:
You can turn this into a dictionary of module to args and use that to init based on a key that indicates the name of a benchmark.

So you can do
```
python benchmark.py --benchmark relu
```
or such. and make "relu" an argument.